jar/jar.go

```diff
@@ -223,29 +251,55 @@ func (c *checker) done() bool {
 	return c.bad() && c.mainClass != ""
 }
 
-func (c *checker) bad() bool {
-	// Care must be taken in the formulae below with respect to the
-	// !c.hasIsJndiEnabled clause.  It is satisfied by default until
-	// JndiManager.class is encountered.  To prevent early termination of a
-	// scan with an incorrect result, we have to ensure that we have already
-	// encountered JndiManager.class (e.g. hasJndiManager*) or we have
-	// encountered positive evidence that it will be absent
-	// (i.e. log4j <2.1).
-
+// Vulnerability signatures.
+// Note: Care must be taken in the formulae below with respect to the
+// !c.hasIsJndiEnabled clause.  It is satisfied by default until
+// JndiManager.class is encountered.  To prevent early termination of
+// a scan with an incorrect result, we have to ensure that we have
+// already encountered JndiManager.class (e.g. hasJndiManager*) or we
+// have encountered positive evidence that it will be absent
+// (i.e. log4j <2.1).
+var sigs = map[cveID]func(*checker) bool{
 	// CVE-2021-44228 - Initial log4shell vulnerability affecting
 	// Log4j2 2.0-beta9 through 2.12.1 (inclusive, 2.12.2 is not
 	// vulnerable) and 2.13.0 through 2.15.0 (exclusive).
-	vulnerablePre215 := c.hasLookupClass && // unpatched >=2.0-beta9 and
-		(c.hasInitialContext || // <2.1
-			c.hasJndiManagerPre215) && // >=2.1 && <2.15 and
-		!c.hasIsJndiEnabled // <2.16 && !2.12.2
+	cve_2021_44228: func(c *checker) bool {
+		return c.hasLookupClass && // unpatched >=2.0-beta9 and
+			(c.hasInitialContext || // <2.1
+				c.hasJndiManagerPre215) && // >=2.1 && <2.15 and
+			!c.hasIsJndiEnabled // <2.16 && !2.12.2
+	},
+
+	// CVE-2021-45046 - Thread Context Lookup Pattern
+	// vulnerability affects all Log4j2 versions >=2.0-beta9 and
+	// <=2.15.0, except for 2.12.2.
+	// See: https://logging.apache.org/log4j/2.x/security.html
+	cve_2021_45046: func(c *checker) bool {
+		return c.hasLookupClass && // unpatched >=2.0-beta9 and
+			(c.hasInitialContext || // <2.1
```

Comment:
Question: Is the extra check for `hasInitialContext` intentional? It was not included in the original check.